app/assets/javascripts/discourse/app/components/bookmark-menu.gjs

```diff
@@ -79,22 +80,25 @@ export default class BookmarkMenu extends Component {
   }
 
   @action
-  async onBookmark() {
-    try {
-      await this.bookmarkManager.create();
-      // We show the menu with Edit/Delete options if the bokmark exists,
-      // so this "quicksave" will do nothing in that case.
-      // NOTE: Need a nicer way to handle this; otherwise as soon as you save
-      // a bookmark, it switches to the other Edit/Delete menu.
-      this.quicksaved = true;
-      this.toasts.success({
-        duration: 3000,
-        views: ["mobile"],
-        data: { message: I18n.t("bookmarks.bookmarked_success") },
+  onBookmark() {
```

Comment:
I'm curious... why remove the `async` and go back to the legacy `.then()` syntax?

If I'm reading it correctly, the only functional change here was to add a `.finally()`? That could've been done like


```
try {
  ...
} catch (error) {
  ...
} finally {
  ...
}
```

Reply:
Yes I noticed it, didn't want to nitpick. But you are right, maybe martin doesn't know about the finally in this case.

Reply:
@davidtaylorhq @jjaffeux I knew about the async try/catch/finally, I just wasn't sure how to store the promise of `create()` in this case for later use? For example below, if I assign `something` I thought that would be the result of `create()` not the promise itself?

```javascript
try {
  const something = await this.bookmarkManager.create();
}...
```

Reply:
Since inside `onChooseReminderOption` I need to wait for this earlier `create` promise to complet

Reply:
Ah I see :+1:. One option would be to split across multiple lines like:

```js
const promise = this.bookmarkManager.create();
const promiseResult = await something;
```

Reply:
Ah right that's interesting I didn't know you could do that, TIL :)